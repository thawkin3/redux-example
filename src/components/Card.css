.card {
  margin: 1em;
}

.card-info {
  display: inline-block;
  margin-left: 10px;
  vertical-align: top;
  text-align: left;
}

.card-img {
  width: 75px;
}

.card-link {
  font-size: 1.25em;
  font-weight: bold;
}
