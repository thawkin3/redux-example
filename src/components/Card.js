import React from 'react';
import './Card.css';

const Card = (props) => (
    <div className="card">
        <img className="card-img" src={props.avatar_url} alt="" />
        <div className="card-info">
            <div className="card-link">
                <a href={props.html_url} target="_blank">{props.name}</a>
            </div>
            <div className="card-company">{props.company}</div>
        </div>
    </div>
);

Card.displayName = 'Card';

export default Card;
