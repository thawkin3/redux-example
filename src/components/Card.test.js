import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Card from './Card';

let wrapper;
let props;

describe('Card', () => {
    beforeEach(() => {
        props = {
            avatar_url: 'url/here/img.jpg',
            html_url: 'https://github.com/thawkin3',
            name: 'Tyler Hawkins',
            company: '@instructure',
        };
        wrapper = shallow(<Card {...props} />);
    });

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card {...props} />, div);
    });

    it('matches the snapshot', () => {
        expect(shallowToJson(wrapper)).toMatchSnapshot();
    });

    it('renders only one Card', () => {
        expect(wrapper.length).toEqual(1);
    });

    it('renders elements with classes for card, card-img, card-info, card-link, and card-company', () => {
        expect(wrapper.find('.card').length).toEqual(1);
        expect(wrapper.find('.card-img').length).toEqual(1);
        expect(wrapper.find('.card-info').length).toEqual(1);
        expect(wrapper.find('.card-link').length).toEqual(1);
        expect(wrapper.find('.card-company').length).toEqual(1);
    });
});
