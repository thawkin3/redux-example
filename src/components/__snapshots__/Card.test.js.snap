// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches the snapshot 1`] = `
<div
  className="card"
>
  <img
    alt=""
    className="card-img"
    src="url/here/img.jpg"
  />
  <div
    className="card-info"
  >
    <div
      className="card-link"
    >
      <a
        href="https://github.com/thawkin3"
        target="_blank"
      >
        Tyler Hawkins
      </a>
    </div>
    <div
      className="card-company"
    >
      @instructure
    </div>
  </div>
</div>
`;
